### PR TITLE
Tweak: Kit-Library - Sorting

### PR DESCRIPTION
**Pull request by Nevoss**
ref: ED-2597

### CODE REVIEW

File: core/app/modules/kit-library/assets/js/components/kit-list.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { CssGrid } from '@elementor/app-ui';
 
 export default function KitList( props ) {
 	return (
-		<CssGrid spacing={ 24 } colMinWidth={ 250 }>
+		<CssGrid spacing={ 24 } colMinWidth={ 250 } >
```
Comment (StyleShit) [inline core/app/modules/kit-library/assets/js/components/kit-list.js:7]: Redundant space

File: core/app/modules/kit-library/assets/js/pages/index/index.scss [hunk 2]
```diff
@@ -6,13 +6,27 @@ $e-kit-library-sticky-search-z-index: 2;
...
+			min-width: 265px;
+		}
+
+		&-sort {
+			min-width: 265px;
+
+			&--safari-support {
+				@include margin-start( spacing( 24 ) );
+			}
+		}
```
Comment (StyleShit) [inline core/app/modules/kit-library/assets/js/pages/index/index.scss:29]: Maybe combine the `min-width` so when someone in the future will change this property, he won't need to go and search for the second occurrence, both of them will get changed.

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: core/app/modules/kit-library/assets/js/pages/index/index.scss (hunk 2 above)
Comment (StyleShit) [reply core/app/modules/kit-library/assets/js/pages/index/index.scss:29]:
```CSS
		&-search,
		&-sort {
			min-width: 265px;
		}

		&-search {
			flex: 1;
		}

		&-sort--safari-support {
			@include margin-start( spacing( 24 ) );
		}
```

**Review by StyleShit — APPROVED** [review]

File: core/app/assets/js/ui/molecules/search-input.scss [hunk 3]
```diff
@@ -6,6 +6,8 @@
 	background: tints(100);
 	outline: none;
 	color: tints(700);
+	line-height: type( line-height, 'flat' );
+	height: 42px;
```
Comment (matipojo) [inline core/app/assets/js/ui/molecules/search-input.scss:10]: Use an existing unit.

File: core/app/assets/js/ui/organisims/sort-select.scss [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	&__select {
+		width: 100%;
+		padding: spacing(10) spacing(10);
+		border: none;
+		background: tints(100);
+		outline: none;
+		color: tints(700);
+		line-height: type( line-height, 'flat' );
+		cursor: pointer;
+		height: 42px;
```
Comment (matipojo) [inline core/app/assets/js/ui/organisims/sort-select.scss:34]: As above

File: core/app/modules/kit-library/assets/js/hooks/use-kits.js [hunk 5]
```diff
@@ -88,3 +96,11 @@ function taxonomiesFilter( item, taxonomies ) {
 		item.taxonomies.some( ( itemTaxonomy ) => taxonomy === itemTaxonomy )
 	);
 }
+
+function sortData( item1, item2, sortValue ) {
```
Comment (matipojo) [inline core/app/modules/kit-library/assets/js/hooks/use-kits.js:100]: Move up

File: core/app/modules/kit-library/assets/js/models/kit.js [hunk 6]
```diff
@@ -35,8 +35,8 @@ export default class Kit extends BaseModel {
 			trendIndex: kit.trend_index,
 			popularityIndex: kit.popularity_index,
 			featuredIndex: kit.featured_index,
-			createdAt: new Date( kit.created_at ),
-			updatedAt: new Date( kit.updated_at ),
+			createdAt: kit.created_at ? new Date( kit.created_at ) : null,
+			updatedAt: kit.updated_at ? new Date( kit.updated_at ) : null,
```
Comment (matipojo) [inline core/app/modules/kit-library/assets/js/models/kit.js:39]: Add TODO to remove on the API is stable

File: core/app/modules/kit-library/assets/js/pages/index/index.js [hunk 7]
```diff
@@ -99,14 +101,28 @@ export default function Index() {
...
+					</Grid>
+					<Grid
+						item
+						className="e-kit-library__index-layout-top-area-sort"
+					>
+						<SortSelect
+							options={ [
+								{ label: 'New', value: 'createdAt' },
+								{ label: 'Popular', value: 'popularityIndex' },
+								{ label: 'Trending', value: 'trendIndex' },
```
Comment (matipojo) [inline core/app/modules/kit-library/assets/js/pages/index/index.js:113]: Add a translations test.

File: core/app/modules/kit-library/data/repository.php [hunk 8]
```diff
@@ -167,6 +167,10 @@ private function transform_kit_api_response( $kit, $manifest = null ) {
 				'access_level' => $kit->access_level,
 				'keywords' => $kit->keywords,
 				'taxonomies' => $taxonomies,
+				'trend_index' => isset( $kit->trend_index ) ? $kit->trend_index : 0,
+				'popularity_index' => isset( $kit->popularity_index ) ? $kit->popularity_index : 0,
+				'created_at' => isset( $kit->created_at ) ? $kit->created_at : null,
+				'updated_at' => isset( $kit->updated_at ) ? $kit->updated_at : null,
```
Comment (matipojo) [inline core/app/modules/kit-library/data/repository.php:173]: TODO

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**